### DISCUSSION TITLE
Skip running tests from Nebula Logger Unlocked package

### DISCUSSION THREAD

**Discussion by gauravchaube21** [jongpie/NebulaLogger#621] [Q&A] [answered]
We have installed Nebula Logger as an unlocked package in our org. Our pipeline installs the Nebula Logger to a scratch org whenever it gets created. Along with the package, we install the regular metadata and run test classes. This happens via our CI/CD Pipeline. We use the following coming to run the test classes
sf apex run test --test-level RunLocalTests --wait 30 --code-coverage --detailed-coverage --result-format human

Unfortunately, even though we have specified `RunLocalTests` in our sf command, all of the test classes run when running the command. Infact, a lot of test classes fail (Error Message: `Search Unavailable`) when this happens. 
![image (2)](https://github.com/jongpie/NebulaLogger/assets/133253065/3bd46087-990c-48cc-b878-4fb975e21502)

It would be great if someone could help us with the reason why the package test classes run even though we explicitly run local test classes and a possible workaround in this situation. Our pipelines are blocked because of the failing test classes within the package.

Thanks in advance

**Comment by jongpie** [comment]
Hi @gauravchaube21 - running Nebula Logger's tests is unfortunately expected behavior when using the unlocked package and `--test-level RunLocalTests`. In this case, the term 'local' refers to any Apex classes without a namespace - and Nebula Logger's unlocked package does not have a namespace, so it's considered 'local code', even though it comes from an unlocked package. 

There was a similar question about this last year (see [my comment here](https://github.com/jongpie/NebulaLogger/issues/478#issuecomment-1501019732)) - one option is for you to create a test suite containing your tests (and excluding Nebula Logger's tests). It's not really ideal, you'd have to maintain a test suite, but it's one way to avoid running Nebula Logger's tests.

As far as that specific error about 'Search unavailable' - I've been running into that a lot myself over the last few weeks, I think it might be related to some Salesforce infrastructure issues. I'm going to see if there's anything I can do in that test to avoid this issue - this single test method isn't that important, it shouldn't cause pipeline failures.

**Reply by gauravchaube21** [reply]
Thank you so much for your response, @jongpie. I really appreciate your suggestions for the issue. I'll try setting up a test suite for the time being. Meanwhile, do you think you would be able to pinpoint the test class problem and patch that out in the package?

**Reply by jongpie** [reply]
@gauravchaube21 I'm still exploring options for how to solve this - it's throwing an `UnexpectedException`, which makes this difficult to handle. I don't think these exceptions can be caught in `try-catch` block (Apex will still throw the exception), and I'd prefer not to eliminate the test method (because it provides valuable automated testing.... when it works 🙈 ). 

So, I'm not sure what good options there are for handling this, but I'll try to explore some more options ASAP. If I can find a good approach, I'll release a new version with a fix.

**Reply by jongpie** [reply]
@gauravchaube21 I've created issue #624 to track this problem.

**Reply by jongpie** [reply, marked as answer]
@gauravchaube21 this has been "fixed" in [release `v4.13.0`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.13.0) - I decided to just comment-out the problematic test method 😅 

I'll revisit this later, but the functionality being tested has not changed in a few years, so commenting-out the test method seems safe enough for now. Let me know if you run into any other issues!

**Reply by gauravchaube21** [reply]
Thank you so much @jongpie . I really appreciate your help on this.❤️

**Reply by jongpie** [reply]
No problem!
